File: easy/283. Move Zeroes/283. Move Zeroes.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        for i in range(len(nums)):
            if nums[i] == 0:
                nums = nums[:i] + nums[i + 1 :]  # + [-1]
                # nums.append(nums[i])


solution = Solution()

case_1 = [0, 1, 0, 3, 12]
solution.moveZeroes(case_1)
print(case_1)

File: easy/283. Move Zeroes/283. Move Zeroes.go
package main

func moveZeroes(nums []int) {
	left := 0

	for right := 0; right < len(nums); right++ {
		if nums[right] != 0 {
			nums[right], nums[left] = nums[left], nums[right]
			left += 1
		}
	}
}
